core/src/main/scala/kafka/consumer/ZookeeperConsumerConnector.scala

```diff
@@ -696,9 +696,9 @@ private[kafka] class ZookeeperConsumerConnector(val config: ConsumerConfig,
             if (topicRegistry.size == 0)
               new java.util.HashMap[String, java.util.Set[java.lang.Integer]]
             else
-              mapAsJavaMap(topicRegistry.map(topics =>
-                topics._1 -> topics._2.keys
-              ).toMap).asInstanceOf[java.util.Map[String, java.util.Set[java.lang.Integer]]]
+              topicRegistry.map(topics =>
+                topics._1 -> topics._2.keys.asInstanceOf[mutable.Set[Integer]].asJava
```

Comment:
this is actually a bug fix to KAFKA-2284 and is the cause for the test failure for this PR as the test incorrectly verifies the resut is a scala `Set`

I have separated out the bugfix into https://github.com/apache/kafka/pull/2102 so that PR will need to be merged first.